mergify_engine/__init__.py

```diff
@@ -48,3 +55,64 @@ def fixup_sentry_reporting(event, hint):  # pragma: no cover
         before_send=fixup_sentry_reporting,
         integrations=[CeleryIntegration(), FlaskIntegration(), RedisIntegration()],
     )
+
+
+RETRY = urllib3.Retry(
+    total=None,
+    redirect=3,
+    connect=5,
+    read=5,
+    status=5,
+    backoff_factor=0.2,
+    status_forcelist=list(range(500, 599)) + [429],
+    method_whitelist=[
+        "HEAD",
+        "TRACE",
+        "GET",
+        "PUT",
+        "OPTIONS",
+        "DELETE",
+        "POST",
+        "PATCH",
+    ],
+    raise_on_status=False,
+)
+
+
+SINGLE_PULL_API_RE = re.compile(
+    f"^https://api.{config.GITHUB_DOMAIN}:443/repos/[^/]+/[^/]+/pulls/\\d+$"
```

Comment:
I'd be good to have a few unit tests around this one just in case…

Reply:
done